aws-go-sdk-dynamodb-example.go

```diff
@@ -0,0 +1,139 @@
+// This program creates a table in dynamodb, puts an item (INSERT),
+// Update the item (adding new attributes (columns))
+
+package main
+
+import (
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+)
+
+func main() {
+
+	// Create a new instance of the DynamoDB service client
+	svc := dynamodb.New(session.New(&aws.Config{Region: aws.String("us-west-2")}))
+
+	tableName := "AWS_SDK_TEST_TABLE"
+
+	// Use CreateTable API Operation to create a table on DynamoDB
+	log.Println("Creating table: ", tableName)
+	if _, err := svc.CreateTable(&dynamodb.CreateTableInput{
+		TableName: aws.String(tableName),
+		AttributeDefinitions: []*dynamodb.AttributeDefinition{
+			{
+				AttributeName: aws.String("Filename"),
+				AttributeType: aws.String(dynamodb.ScalarAttributeTypeS),
+			},
+		},
+		KeySchema: []*dynamodb.KeySchemaElement{
+			{
+				AttributeName: aws.String("Filename"),
+				KeyType:       aws.String("HASH"),
+			},
+		},
+		ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+			ReadCapacityUnits:  aws.Int64(1),
+			WriteCapacityUnits: aws.Int64(1),
+		},
+	}); err != nil {
+		log.Println("Failed to create Amazon DynamoDB table,", err)
+		return
+	}
+	log.Println("Succesfully created table:", tableName, "\n")
+
+	// Now we will list the tables and will see if the table
+	// that we created above is printed or not
+	log.Println("Getting tablenames from DynamoDB")
+	result, err := svc.ListTables(&dynamodb.ListTablesInput{})
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	log.Println("Tables:")
+	for i, table := range result.TableNames {
+		log.Println("\t", i+1, *table)
+	}
+	log.Println()
+
+	// If we create a table and immediately Put data, we get
+	// a HTTP 400 error. So sleep for 5 seconds and try
+	time.Sleep(5 * time.Second)
```

Comment:
It would be great to update this section to use the SDK's waiters. `DynamoDB.WaitUntilTableExists` will provide this functionality and will return when the table is created. Which is great if it were to take more than 5 seconds in this case.